### PR TITLE
Feature/daily txs volume chart page

### PR DESCRIPTION
**Pull request by BBCmisha**

### CODE REVIEW

File: src/helpers/customChartHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    if (rest.fee) {
+      const amount = Number(rest.fee[0]?.amount) || 0
+      values.push(amount)
+    } else {
+      values.push(Object.values(rest)[0])
+    }
+  })
+
+  return { data: values, labels }
+}
```
Comment (lukachi) [inline src/helpers/customChartHelpers.ts:207]: need to discuss

File: src/views/AverageBlockTimeChart.vue [hunk 2]
```diff
@@ -0,0 +1,124 @@
+<template>
```
Comment (lukachi) [inline src/views/AverageBlockTimeChart.vue:1]: Check Bem in this file

File: src/views/AverageBlockTimeChart.vue [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import { Router, useRouter } from 'vue-router'
+import { callers } from '@/api/callers'
+import { handleError } from '@/helpers/errors'
+import { formatDataForCharts } from '@/helpers/customChartHelpers'
+import { sortingDaysForChart } from '@/helpers/helpers'
+import CustomBarChart from '@/components/Charts/CustomBarChart.vue'
+import BackButton from '@/components/BackButton.vue'
+
+export default defineComponent({
+  name: 'AverageBlockTimeChart',
```
Comment (lukachi) [inline src/views/AverageBlockTimeChart.vue:53]: Check all component names, be sure they are all in kebab case

File: src/views/AverageBlockTimeChart.vue [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        chartData.value = formatDataForCharts(data)
+      } catch (error) {
+        handleError(error as Error)
+        console.error(error)
+      } finally {
+        isLoading.value = false
+      }
+    }
+
+    watch(sortingValue, async (): Promise<void> => await getChartData())
```
Comment (lukachi) [inline src/views/AverageBlockTimeChart.vue:79]: maybe call it on vue-picker @input event?

File: src/views/AverageOdinBlockSizeChart.vue [hunk 5]
```diff
@@ -0,0 +1,124 @@
+<template>
```
Comment (lukachi) [inline src/views/AverageOdinBlockSizeChart.vue:1]: BEM

File: src/views/AverageOdinBlockSizeChart.vue [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        chartData.value = formatDataForCharts(data)
+      } catch (error) {
+        handleError(error as Error)
+        console.error(error)
+      } finally {
+        isLoading.value = false
+      }
+    }
+
+    watch(sortingValue, async (): Promise<void> => await getChartData())
```
Comment (lukachi) [inline src/views/AverageOdinBlockSizeChart.vue:79]: meybe call in @input event?

File: src/views/AverageTransactionFeeChart.vue [hunk 7]
```diff
@@ -0,0 +1,124 @@
+<template>
```
Comment (lukachi) [inline src/views/AverageTransactionFeeChart.vue:1]: Bem

File: src/helpers/customChartHelpers.ts [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+        display: false,
+      },
+      tooltip: {
+        yAlign: 'bottom',
+        padding: 0,
+        enabled: false,
+        external: function (ctx): void {
+          const tooltipModel = ctx.tooltip
+          let tooltipEl = document.getElementById('chartjs-tooltip')
+          let tooltipArrow = tooltipEl?.querySelector('.arrow') as HTMLElement
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/helpers/customChartHelpers.ts:43]: this class is added by the chartJs library

File: src/api/callers.ts [hunk 9]
```diff
@@ -172,6 +173,50 @@ const makeCallers = () => {
       return getAPIDate(`${API_CONFIG.rpc}/tx?hash=0x${hash}&prove=true`)
     },
     getParams: querier((qc) => qc.mint.unverified.params),
+    getTxVolumePerDays: (start_time: Date, end_time: Date) => {
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/api/callers.ts:176]: done

File: src/api/callers.ts [hunk 10]
```diff
@@ -172,6 +173,50 @@ const makeCallers = () => {
...
+      return axios.get(
+        `${API_CONFIG.telemetryUrl}/telemetry/blocks/avgTxFeePerDays`,
+        {
+          params: {
+            start_time: (start_time.getTime() / 1000).toFixed(),
+            end_time: (end_time.getTime() / 1000).toFixed(),
+          },
+        }
+      )
+    },
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/api/callers.ts:219]: done

File: src/components/Charts/CustomBarChart.vue [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+Chart.register(...registerables)
+
+export default defineComponent({
+  components: { BarChart },
+  props: {
+    datasetLabel: { type: String, default: 'Dataset' },
+    datasetUnit: { type: String, default: '' },
+    data: {
+      type: (Object as PropType<formatedTelemetryDataForCharts>) || undefined,
+    },
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/components/Charts/CustomBarChart.vue:21]: done

File: src/components/Charts/CustomBarChart.vue [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+  },
+  setup: function (props) {
+    const _data = toRef(props, 'data')
+    const _datasetLabel = toRef(props, 'datasetLabel')
+    const _datasetUnit = toRef(props, 'datasetUnit')
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/components/Charts/CustomBarChart.vue:26]: done

File: src/components/Charts/CustomLineChart.vue [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+Chart.register(...registerables)
+
+export default defineComponent({
+  components: { LineChart },
+  props: {
+    datasetLabel: { type: String, default: 'Dataset' },
+    datasetUnit: { type: String, default: '' },
+    data: {
+      type: Object as PropType<formatedTelemetryDataForCharts>,
+    },
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/components/Charts/CustomLineChart.vue:23]: done

File: src/components/Charts/CustomLineChart.vue [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+  },
+  setup: function (props) {
+    const _data = toRef(props, 'data')
+    const _datasetLabel = toRef(props, 'datasetLabel')
+    const _datasetUnit = toRef(props, 'datasetUnit')
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/components/Charts/CustomLineChart.vue:28]: done

File: src/components/InfoPanel/InfoPanel.vue [hunk 15]
```diff
@@ -26,136 +23,50 @@
...
-
       const endDate = new Date()
       const startDate = new Date()
-      startDate.setDate(startDate.getDate() - 2)
-
-      const { txVolumePerDay } = await callers.getTxVolume({
-        startDate,
-        endDate,
-      })
+      startDate.setDate(startDate.getDate() - 4)
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/components/InfoPanel/InfoPanel.vue:61]: done

File: src/components/Nav.vue [hunk 16]
```diff
@@ -108,6 +84,4 @@ export default defineComponent({
 })
 </script>
 
-<style scoped lang="scss">
-
-</style>
+<style scoped lang="scss"></style>
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/components/Nav.vue:87]: done

File: src/helpers/customChartHelpers.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/helpers/customChartHelpers.ts:43]: done

File: src/styles/custom.scss [hunk 17]
```diff
@@ -5,9 +5,39 @@ body {
...
 
+#chartjs-tooltip {
+  position: absolute;
+  padding: 1.2rem 1.6rem;
+  pointer-events: none;
+  background-color: #053F7D;
+  border-radius: 0.8rem;
+  color: #fff;
+
+  &.bottom {
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/styles/custom.scss:16]: done

File: src/styles/custom.scss [hunk 18]
```diff
@@ -5,9 +5,39 @@ body {
...
+    .arrow {
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/styles/custom.scss:17]: done

File: src/styles/custom.scss [hunk 19]
```diff
@@ -5,9 +5,39 @@ body {
...
+      display: block;
+      width: 14px;
+      height: 14px;
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/styles/custom.scss:20]: done

File: src/styles/custom.scss [hunk 20]
```diff
@@ -5,9 +5,39 @@ body {
...
+      position: absolute;
+      background-color: #053F7D;
+      transform: translateX(-7px) rotate(45deg);
+      bottom: -5px;
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/styles/custom.scss:24]: done

File: src/views/AverageBlockTimeChart.vue (hunk 2 above)
Comment (BBCmisha) [reply src/views/AverageBlockTimeChart.vue:1]: done

File: src/views/AverageBlockTimeChart.vue (hunk 3 above)
Comment (BBCmisha) [reply src/views/AverageBlockTimeChart.vue:53]: done

File: src/views/AverageBlockTimeChart.vue [hunk 21]
```diff
@@ -0,0 +1,124 @@
...
+      chartData,
+      isLoading,
+      sortingValue,
+      sortingDaysForChart,
+    }
+  },
+})
+</script>
+
+<style lang="scss" scoped>
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/AverageBlockTimeChart.vue:96]: done

File: src/views/AverageOdinBlockSizeChart.vue (hunk 5 above)
Comment (BBCmisha) [reply src/views/AverageOdinBlockSizeChart.vue:1]: done

File: src/views/AverageTransactionFeeChart.vue (hunk 7 above)
Comment (BBCmisha) [reply src/views/AverageTransactionFeeChart.vue:1]: done

File: src/views/AverageTransactionFeeChart.vue [hunk 22]
```diff
@@ -0,0 +1,124 @@
...
+import { Router, useRouter } from 'vue-router'
+import { callers } from '@/api/callers'
+import { handleError } from '@/helpers/errors'
+import { formatDataForCharts } from '@/helpers/customChartHelpers'
+import { sortingDaysForChart } from '@/helpers/helpers'
+import CustomLineChart from '@/components/Charts/CustomLineChart.vue'
+import BackButton from '@/components/BackButton.vue'
+
+export default defineComponent({
+  name: 'AverageTransactionFeeChart',
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/AverageTransactionFeeChart.vue:53]: done

File: src/views/AverageTransactionFeeChart.vue [hunk 23]
```diff
@@ -0,0 +1,124 @@
...
+      startDate.setDate(startDate.getDate() - Number(sortingValue.value))
+
+      isLoading.value = true
+      try {
+        const { data } = await callers.getAvgTxFeePerDays(startDate, endDate)
+
+        chartData.value = formatDataForCharts(data)
+      } catch (error) {
+        handleError(error as Error)
+        console.error(error)
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/AverageTransactionFeeChart.vue:73]: done

File: src/views/AverageTransactionFeeChart.vue [hunk 24]
```diff
@@ -0,0 +1,124 @@
...
+      isLoading,
+      sortingValue,
+      sortingDaysForChart,
+    }
+  },
+})
+</script>
+
+<style lang="scss" scoped>
+.view-title {
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/AverageTransactionFeeChart.vue:97]: done

File: src/views/Charts-stats.vue [hunk 25]
```diff
@@ -0,0 +1,146 @@
+<template>
+  <div class="container">
+    <div class="mg-b32 mg-t32">
+      <h2 class="view-title">Charts & Stats</h2>
+    </div>
+    <div class="charts-section">
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/Charts-stats.vue:6]: done

File: src/views/Charts-stats.vue [hunk 26]
```diff
@@ -0,0 +1,146 @@
...
+    </div>
+  </div>  
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+import { Router, useRouter } from 'vue-router'
+
+export default defineComponent({
+  name: 'Chart-stats',
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/Charts-stats.vue:37]: done

File: src/views/Charts-stats.vue [hunk 27]
```diff
@@ -0,0 +1,146 @@
...
+      redirectToChartPage,
+    }
+  },
+})
+</script>
+
+<style lang="scss" scoped>
+.charts-section {
+  padding: 3.2rem 2.4rem;
+  border: 1px solid var(--clr__action);
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/Charts-stats.vue:84]: done

File: src/views/Charts-stats.vue [hunk 28]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  &__table {
+    width: 100%;
+    border-collapse: collapse;
+    border-style: hidden;
+
+    td {
+      width: 33.3333%;
+      border: 1px solid var(--clr__table-border);
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/Charts-stats.vue:100]: done

File: src/views/Charts-stats.vue [hunk 29]
```diff
@@ -0,0 +1,146 @@
...
+      cursor: pointer;
+
+      &:hover {
+        // background-color: var(--clr__table-row-hover);
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/Charts-stats.vue:104]: done

File: src/views/Charts-stats.vue [hunk 30]
```diff
@@ -0,0 +1,146 @@
...
+        background-color: #ecf5ff;
+      }
+    }
+
+    .table-cell {
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/Charts-stats.vue:109]: done

File: src/views/DailyTransactionsVolumeChart.vue [hunk 31]
```diff
@@ -0,0 +1,120 @@
+<template>
+  <div class="container">
+    <div class="page-title">
+      <BackButton :current-router="router" :text="'Charts & Stats'" />
+      <h2 class="view-title">DailyTransactionsVolumeChart</h2>
+    </div>
+
+    <div class="sort-wrapper">
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/DailyTransactionsVolumeChart.vue:8]: done

File: src/views/DailyTransactionsVolumeChart.vue [hunk 32]
```diff
@@ -0,0 +1,120 @@
...
+import { Router, useRouter } from 'vue-router'
+import { callers } from '@/api/callers'
+import { handleError } from '@/helpers/errors'
+import { formatDataForCharts } from '@/helpers/customChartHelpers'
+import { sortingDaysForChart } from '@/helpers/helpers'
+import CustomLineChart from '@/components/Charts/CustomLineChart.vue'
+import BackButton from '@/components/BackButton.vue'
+
+export default defineComponent({
+  name: 'DailyTransactionsVolumeChart',
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/DailyTransactionsVolumeChart.vue:49]: done

File: src/views/DailyTransactionsVolumeChart.vue [hunk 33]
```diff
@@ -0,0 +1,120 @@
...
+      startDate.setDate(startDate.getDate() - Number(sortingValue.value))
+
+      isLoading.value = true
+      try {
+        const { data } = await callers.getTxVolumePerDays(startDate, endDate)
+
+        chartData.value = formatDataForCharts(data)
+      } catch (error) {
+        handleError(error as Error)
+        console.error(error)
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/DailyTransactionsVolumeChart.vue:69]: done

File: src/views/DailyTransactionsVolumeChart.vue [hunk 34]
```diff
@@ -0,0 +1,120 @@
...
+      chartData,
+      isLoading,
+      sortingValue,
+      sortingDaysForChart,
+    }
+  },
+})
+</script>
+
+<style lang="scss" scoped>
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/views/DailyTransactionsVolumeChart.vue:92]: done

File: src/views/AverageOdinBlockSizeChart.vue (hunk 6 above)
Comment (BBCmisha) [reply src/views/AverageOdinBlockSizeChart.vue:79]: there is no input event in this custom component

File: src/views/AverageBlockTimeChart.vue (hunk 4 above)
Comment (BBCmisha) [reply src/views/AverageBlockTimeChart.vue:79]: there is no input event in this custom component

File: src/components/SearchBar/BlockResultItem.vue [hunk 35]
```diff
@@ -23,8 +23,7 @@
             :to="`/transactions/${result?.block?.header?.height}`"
             class="app-table__cell-txt"
             :text="`${cropText(
-              '0x' +
-                toHexFunc(result?.block?.header?.validatorsHash).toUpperCase()
+              toHexFunc(result?.block?.header?.validatorsHash).toUpperCase()
```
[the remark this answers is not in the document]
Comment (BBCmisha) [reply src/components/SearchBar/BlockResultItem.vue:26]: done

**Pull request merged**